src/network/networkd-address.c

```diff
@@ -495,8 +495,9 @@ static int address_acquire(Link *link, Address *original, Address **ret) {
         assert(ret);
 
         /* Something useful was configured? just use it */
-        if (in_addr_is_null(original->family, &original->in_addr) <= 0)
-                return 0;
+        r = in_addr_is_null(original->family, &original->in_addr);
+        if (r <= 0)
+                return r;
```

Comment:
hmm, this function logs about all other errors. this should log about the r < 0 cases here too, no?

Reply:
hmm, actually it's chaos right now. This should probably be cleared up, but probably doesn't have to be in this PR. But in the long run we should make sure that each function either logs about all errors or none, and this one is a bit messy there right now, as it logs about failures to acquire addresses from the pool, but not on -ENOMEM for example...

but anyway, given that this patch doesn't really make things worse, let's ignore this for this PR

Reply:
Theoretically yes. I hope that the family is already checked at somewhere before calling this function. But yes, I will add the logs.